test/src/org/exist/storage/ModificationTimeTest.java

```diff
@@ -0,0 +1,209 @@
+package org.exist.storage;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.exist.collections.triggers.TriggerException;
+import org.exist.security.PermissionDeniedException;
+import org.exist.util.LockException;
+import org.junit.BeforeClass;
+import org.exist.dom.persistent.BinaryDocument;
+import org.exist.EXistException;
+import org.exist.xmldb.XmldbURI;
+import org.exist.test.TestConstants;
+import org.exist.collections.Collection;
+import org.exist.Database;
+import org.exist.collections.IndexInfo;
+import org.exist.dom.persistent.DocumentImpl;
+import org.exist.start.Main;
+import org.exist.storage.txn.TransactionManager;
+import org.exist.storage.txn.Txn;
+import org.exist.util.Configuration;
+import org.exist.util.FileUtils;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.exist.util.ConfigurationHelper;
+import org.exist.util.DatabaseConfigurationException;
+import org.junit.After;
+import org.xml.sax.SAXException;
+
+
+public class ModificationTimeTest {
+
+    protected static String XML_FILENAME = "test.xml";
+    protected static String VALID_XML = "<?xml version=\"1.0\"?>" +
+                                        "<valid/>";
+    protected static String INVALID_XML = "<?xml version=\"1.0\"?>"
+            + "<invalid>";
+
+    
+    @BeforeClass
+    public static void ensureCleanDatabase() throws IOException {
```

Comment:
It seems to me that this would be useful for other tests too.
Could you move the logic of this method into org.exist.TestUtils?